src/Microsoft.PowerShell.Commands.Utility/commands/utility/MatchString.cs

```diff
@@ -1518,6 +1628,18 @@ private bool DoMatchWorker(string operandString, MatchInfo matchInfo, out MatchI
             int patternIndex = 0;
             matchResult = null;
 
+            List<int> indexes = null;
+            List<int> lengths = null;
+
+            // If Emphasize is set and VT is supported,
+            // the lengths and starting indexes of regex matches
+            // need to be passed in to the matchInfo object.
+            if (Emphasize && Host.UI.SupportsVirtualTerminal)
```

Comment:
Given how many times `Emphasize && Host.UI.SupportsVirtualTerminal` it might be worth saving its result for reuse:

```csharp
bool shouldEmphasize = Emphasize && Host.UI.SupportsVirtualTerminal;

if (shouldEmphasize)
{
    ...
}
```